core/src/main/java/cucumber/runtime/ExitStatus.java

```diff
@@ -24,16 +25,18 @@ public void receive(TestCaseFinished event) {
         }
     };
 
-    ExitStatus() {
+    ExitStatus(RuntimeOptions runtimeOptions) {
+        this.runtimeOptions = runtimeOptions;
     }
 
-
     @Override
     public void setEventPublisher(EventPublisher publisher) {
         publisher.registerHandlerFor(TestCaseFinished.class, testCaseFinishedHandler);
     }
 
-    public byte exitStatus(boolean isStrict) {
-        return results.isEmpty() || max(results, SEVERITY).isOk(isStrict) ? 0x0 : ERRORS;
+    public byte exitStatus() {
+        return results.isEmpty() ||
+            (runtimeOptions.isWip() ? max(results, SEVERITY).isOkWip(runtimeOptions.isStrict()) :
+                max(results, SEVERITY).isOk(runtimeOptions.isStrict())) ? 0x0 : ERRORS;
```

Comment:
It's okay to use if statements here and use [a guard clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) to handle the empty case.

I also think this will fail when there is one test case that passes and one test case that failed.

Reply:
Can you add a test case to prove it works?

Reply:
Yep, I can add test to check this, but… where I write it ?

Reply:
In runtime test. But you put more then one event on the bus.

Reply:
It is (normally) fixed with 99e535e